### DISCUSSION TITLE
Infinispan soft index file store performance issue

### DISCUSSION THREAD

**Discussion by megicivovic** [keycloak/keycloak#14846] [Archive]
Hi all,
we are trying to upgrade from Keycloak 15(Wildfly) to Keycloak X version 18, and also from Infinispan 11 to Infinispan 13. We are experiencing severe performance issues with the change of default file store from single file store to soft index file store. All of the details can be found [here](https://infinispan.zulipchat.com/#narrow/stream/118645-infinispan/topic/Infinispan.2013.20Store.20iteration.20performance)

We are using remote store with persistence on Keycloak (embedded caches) side : 

` <distributed-cache name="sessions" configuration="distributed-cache-template">
            <remote-store cache="sessions" xmlns="urn:infinispan:config:store:remote:13.0"
                          raw-values="true"
                          marshaller="org.keycloak.cluster.infinispan.KeycloakHotRodMarshallerFactory"
                          shared="true" segmented="false"
                          connect-timeout="${env.KEYCLOAK_REMOTE_ISPN_CONN_TIMEOUT:2000}">
                <remote-server host="${env.INFINISPAN_HOST:infinispan}" port="${env.INFINISPAN_PORT:11222}"/>
                <security>
                    <authentication>
                        <digest username="${env.INFINISPAN_USERNAME:keycloak}"
                                password="${env.INFINISPAN_PASSWORD:password}"
                                realm="default"/>
                    </authentication>
                </security>
            </remote-store>
        </distributed-cache>`

Any sort of guidance around configuration or advice would be appreciated!

**Comment by kkcmadhu** [comment]
i too have silmilar issues. https://github.com/keycloak/keycloak/discussions/15174 :( no luck so far!

**Comment by hmlnarik** [comment]
Keycloak legacy store is not tested with infinispan using remote-store. While we thus cannot offer help here, let me recommend your checking with Infinispan team. See https://infinispan.org/community/ for pointers.
